roxie/ccd/ccdquery.cpp

```diff
@@ -106,16 +128,18 @@ class CQueryDll : implements IQueryDll, public CInterface
     }
     static const CQueryDll *getQueryDll(const char *dllName, bool isExe)
     {
-        CriticalBlock b(dllCacheLock);
-        CQueryDll *dll = dllCache.getValue(dllName);
-        if (dll && dll->isAliveAndLink())
-            return dll;
-        else
+        Owned<CQueryDll> dll;
```

Comment:
Using an Owned here isn't correct:  I think you need to go back to using a pointer.

Two threads get to this point with an object from the cache.
Third thread releases the last reference to the object and blocks on the crtical section in beforeDispose
1st thread gets here, sees the object is dead.  That will cause dll to be released  - since it is in an owned
2nd thread gets here, adn object now appears to be alive since link count has wrapped below the dead marker
3rd thread disposes of the object